File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/application/port/outbound/bike/BikeSubscriptionProcess.java
package io.miragon.example.application.port.outbound.bike;

import io.miragon.example.domain.bike.BikeId;
import io.miragon.example.domain.bike.BikeSubscriptionId;

public interface BikeSubscriptionProcess {
    long startSubscription(BikeSubscriptionId id, BikeId bikeId);
    void sendPaymentReceived(BikeSubscriptionId id);
    void sendRequestCanceled(BikeSubscriptionId id);
    void sendBikeReceived(BikeSubscriptionId id);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/application/port/outbound/bike/BikeSubscriptionProcess.kt
package io.miragon.example.application.port.outbound.bike

import io.miragon.example.domain.bike.BikeId
import io.miragon.example.domain.bike.BikeSubscriptionId

interface BikeSubscriptionProcess {
    fun startSubscription(id: BikeSubscriptionId, bikeId: BikeId): Long
    fun sendPaymentReceived(id: BikeSubscriptionId)
    fun sendRequestCanceled(id: BikeSubscriptionId)
    fun sendBikeReceived(id: BikeSubscriptionId)
}
